os/_socket failed to compile on NetBSD

This is on NetBSD 8.0.
```
... compile foreign os/_socket
../../../src/std/os/_socket.c: In function 'ffi_socket_setsockopt_mreq_src':
../../../src/std/os/_socket.c:2437:24: error: storage size of 'mreq' isn't known
  struct ip_mreq_source mreq;
                        ^
../../../src/std/os/_socket.c:2438:27: error: invalid application of 'sizeof' to incomplete type 'struct ip_mreq_source'
  socklen_t olen = sizeof (struct ip_mreq_source);
                           ^
```

I'm not sure if `ip_mreq_source` is defined anywhere on NetBSD.

--- a/src/std/os/_socket.scm
+++ b/src/std/os/_socket.scm
@@ -786,7 +786,7 @@ int ffi_socket_setsockopt_mreq (int fd, int level, int opt, ___SCMOBJ maddr, ___
 
 int ffi_socket_setsockopt_mreq_src (int fd, int level, int opt, ___SCMOBJ maddr, ___SCMOBJ iaddr, ___SCMOBJ saddr)
 {
-#ifndef __OpenBSD__
+#if !defined(__OpenBSD__) && !defined(__NetBSD__)
  struct ip_mreq_source mreq;
  socklen_t olen = sizeof (struct ip_mreq_source);
  memcpy (&mreq.imr_multiaddr, U8_DATA (maddr), sizeof (struct in_addr));

